addon/components/frost-password.js

```diff
@@ -1,103 +1,82 @@
-import _ from 'lodash'
+import computed from 'ember-computed-decorators'
 import Ember from 'ember'
-const {Component} = Ember
-import computed, {readOnly} from 'ember-computed-decorators'
+const {
+  Component,
+  on,
+  run
+} = Ember
+import FrostEventsProxy from '../mixins/frost-events-proxy'
 import layout from '../templates/components/frost-password'
 
-export default Component.extend({
-  // ==========================================================================
-  // Dependencies
-  // ==========================================================================
+export default Component.extend(FrostEventsProxy, {
 
-  // ==========================================================================
-  // Properties
-  // ==========================================================================
+  // == Properties ============================================================
 
-  classNames: ['frost-password'],
-  classNameBindings: ['revealable'],
+  classNames: [
+    'frost-password'
+  ],
+  classNameBindings: [
+    'isCapsOn',
+    'revealable'
+  ],
   isCapsOn: false,
-  isRevealerVisible: false,
+  isRevealed: false,
   layout,
   revealable: false,
-  revealed: false,
-  revealIcon: 'show',
   tabindex: 0,
-  type: 'password',
 
-  // ==========================================================================
-  // Computed Properties
-  // ==========================================================================
+  // == Computed properties  ===================================================
 
-  @readOnly
-  @computed('isCapsOn', 'isRevealerVisible', 'revealable')
-  isCapsAndReveal (isCapsOn, isRevealerVisible, revealable) {
-    return revealable && isCapsOn && isRevealerVisible
+  @computed('isRevealed')
+  revealMessage (isRevealed) {
+    return isRevealed ? 'Hide' : 'Show'
   },
 
-  // ==========================================================================
-  // Functions
-  // ==========================================================================
+  @computed('isRevealed')
+  type (isRevealed) {
+    return isRevealed ? 'text' : 'password'
+  },
 
-  // ==========================================================================
-  // Events
-  // ==========================================================================
+  // == Events ================================================================
 
-  focusOut: Ember.on('focusOut', function () {
+  _focusOut: on('focusOut', function (event) {
+    // this._super(...arguments)
     this.set('isCapsOn', false)
   }),
 
-  keyDown: Ember.on('keyDown', function (e) {
+  _keyDown: Ember.on('keyDown', function (e) {
     var s = e || window.e
-    if (this.get('isCapsOn') === false && (s.which === 20 || s.keyCode === 20)) {
+    if (s.which === 20 || s.keyCode === 20) {
```

Comment:
Rather than using `20` throughout this file can you define a constant at the top of the file and reference that.

```js
const CAPS_LOCK_KEY = 20

…

    if (s.which === CAPS_LOCK_KEY || s.keyCode === CAPS_LOCK_KEY) {
…
```